app/main/db/queries.py

```diff
@@ -288,19 +302,108 @@ def _build_consignment_filter_query(consignment_id: uuid.UUID):
                 ),
                 else_=None,
             ),
-        ),
+        ).label("closure_period"),
     )
 
     filters = [
         File.ConsignmentId == consignment_id,
         func.lower(File.FileType) == "file",
     ]
 
-    query = (
+    sub_query = (
         select.join(FileMetadata, File.FileId == FileMetadata.FileId)
         .filter(*filters)
         .group_by(File.FileId)
         .order_by(File.FileName)
+    ).subquery()
+
+    query = db.session.query(
+        sub_query.c.file_id,
+        sub_query.c.file_name,
+        func.to_char(
+            sub_query.c.date_last_modified,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("date_last_modified"),
+        sub_query.c.closure_type,
+        func.to_char(
+            sub_query.c.closure_start_date,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("closure_start_date"),
+        sub_query.c.closure_period,
     )
 
+    dt_range = get_date_range(date_range)
+    if dt_range["date_from"] is not None and dt_range["date_to"] is not None:
+        query = query.filter(
+            and_(
+                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+                >= dt_range["date_from"],
+                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+                <= dt_range["date_to"],
+            )
+        )
+    elif dt_range["date_from"] is not None:
+        query = query.filter(
+            func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+            >= dt_range["date_from"]
+        )
+    elif dt_range["date_to"] is not None:
+        query = query.filter(
+            func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+            <= dt_range["date_to"]
+        )
+
     return query
+
+
+def _build_date_range_filter_query(query, date_range):
+    dt_range = get_date_range(date_range)
+
+    if dt_range["date_from"] is not None and dt_range["date_to"] is not None:
+        query = query.filter(
+            and_(
+                func.to_char(Consignment.TransferCompleteDatetime, "YYYY-MM-DD")
+                >= dt_range["date_from"],
+                func.to_char(Consignment.TransferCompleteDatetime, "YYYY-MM-DD")
+                <= dt_range["date_to"],
+            )
+        )
+    elif dt_range["date_from"] is not None:
+        query = query.filter(
+            Consignment.TransferCompleteDatetime >= dt_range["date_from"]
+        )
+    elif dt_range["date_to"] is not None:
+        query = query.filter(
+            Consignment.TransferCompleteDatetime <= dt_range["date_to"]
+        )
+    return query
+
+
+def get_date_range(date_range):
```

Comment:
Can we move this to its own date_formatter.py file so that this file is only about queries

Reply:
If we pass in date_from, date_to we could get rid of this function and just do the validation at the top level of the route, which I think makes sense for all input values.

We should validate as quickly as possible, that way we can let a user know if they have entered something incorrect rather than wasting time doing logic that will be wasted.

e.g.
```
def browse():
    form = SearchForm()
    page = int(request.args.get("page", 1))
    per_page = int(current_app.config["DEFAULT_PAGE_SIZE"])
    transferring_body_id = request.args.get("transferring_body_id", None)
    series_id = request.args.get("series_id", None)
    consignment_id = request.args.get("consignment_id", None)

    date_from = # read in and validate from the html template form
    date_to = # read in and validate from the html template form
```

It looks like Flask's WTForms does validation for us out of the box!

https://flask.palletsprojects.com/en/1.1.x/patterns/wtforms/

Reply:
separated logic for get_date_range into date_formatter.py